drivers/bluetooth/hci/hci_silabs_efr32.c

```diff
@@ -155,17 +241,41 @@ static void slz_thread_func(void *p1, void *p2, void *p3)
 	}
 }
 
+static void slz_rx_thread_func(void *p1, void *p2, void *p3)
+{
+	const struct device *dev = DEVICE_DT_GET(DT_DRV_INST(0));
```

Comment:
I think it'd be nice to pass the `struct device` pointer as the first thread parameter (i.e. give it to `k_thread_create()`), so that you don't need to fetch it through these macros anymore.

Reply:
Good point. This part was just moved here, but in its new place your suggestion makes perfect sense.